connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/DataTypeProtoConverter.scala

```diff
@@ -114,6 +118,31 @@ object DataTypeProtoConverter {
     MapType(toCatalystType(t.getKeyType), toCatalystType(t.getValueType), t.getValueContainsNull)
   }
 
+  private def toCatalystUDT(t: proto.DataType.UDT): UserDefinedType[_] = {
+    if (t.getType != "udf") {
```

Comment:
Should it be `udt`?

Reply:
yes, right now this field in all UDTs is `udt`.
but this field is only used in JSON, maybe able to git rid of it.

https://github.com/apache/spark/blob/1a42aa5bd44e7524bb55463bbd85bea782715834/sql/catalyst/src/main/scala/org/apache/spark/sql/types/UserDefinedType.scala#L61-L66

https://github.com/apache/spark/blob/1a42aa5bd44e7524bb55463bbd85bea782715834/sql/catalyst/src/main/scala/org/apache/spark/sql/types/UserDefinedType.scala#L127-L132

https://github.com/apache/spark/blob/1a42aa5bd44e7524bb55463bbd85bea782715834/python/pyspark/sql/types.py#L731-L749

Reply:
Thanks!